react/modules/analytics/utils.ts

```diff
@@ -0,0 +1,95 @@
+// Almost all of the logic of expiration and campaign invalidation is based on this official GA article https://support.google.com/analytics/answer/2731565
+
+export interface AnalyticsData {
+  location: string
+  referrer: string
+  expires: number
+}
+
+export interface AnalyticsDataWithOrigin extends AnalyticsData {
+  origin: 'storage' | 'fresh'
+}
+
+export const UTM_SOURCE_PARAM = 'utm_source'
+export const UTM_MEDIUM_PARAM = 'utm_medium'
+
+// All of these parameters trigger a new session storage of the url
+// This list was inspired by Simo Ahava's list used on his cookie implementation of this feature
+// https://www.simoahava.com/analytics/persist-campaign-data-landing-page/#setup-the-persist-campaign-data-tag
+export const analyticsURLParams = [
+  UTM_SOURCE_PARAM,
+  UTM_MEDIUM_PARAM,
+  'utm_campaign',
+  'gclid',
+  'utm_term',
+  'utm_content',
+  'utm_id',
+] as const
+
+export function shouldInvalidateCurrentCampaign(
+  location: Location,
+  referrer: string
+) {
+  const referrerURL = referrer ? new URL(referrer) : null
+
+  // if user comes from a referring website
+  if (referrerURL && referrerURL.host !== location.host) {
```

Comment:
```suggestion
  if (referrerURL?.host !== location.host) {
```

Reply:
This is not the case. If `referrerURL` is null, the condition should fail. Setting it up like this makes the condition pass because the location host is never null, which is not the desired behavior.

Reply:
Yeah, sorry! You're right